src/helpers/_get-table-name-sandbox.js

```diff
@@ -1,46 +1,48 @@
-// let { join } = require('path')
-// FIXME should export read
 let read = require('@architect/inventory/src/read')
-// let { readArc } = require('@architect/parser')
-
+let tiny = require('tiny-json-http')
 /**
  * Arc 6+ paths where project manifest needs to be read from root
  */
-module.exports = function getTableName () {
-
-  let cwd = process.cwd()
-  let type = 'projectManifest'
+module.exports = async function getTableName () {
   let err = ReferenceError('@tables not defined! begin/data expects a data table')
 
-  try {
-    // attempt to read the arcfile
-    let { arc } = read({ type, cwd })
-
-    // look for a table named 'data'
-    if (arc.tables && arc.tables.some(t => Object.keys(t)[0] === 'data')) {
-      // return the local table
-      return `${arc.app[0]}-staging-data`
+  if (process.env.ARC_ENV === 'testing' ||
+      process.env.NODE_ENV === 'testing') {
+    try {
+      let result = await tiny.get({ url: '/_asd' })
+      let services = result.body
+      if (services.tables.data) {
+        return services.tables.data
+      }
+      else {
+        throw err
+      }
     }
-    else {
-      throw err
+    catch (e) {
+      console.error(e)
+      throw e
     }
   }
-  catch (e) {
-    console.error(e)
-    throw err
-  }
+  else {
+    let cwd = process.cwd()
+    let type = 'projectManifest'
 
-  /*
-  // Try cwd first (usually testing)
-  let result = readArc()
-  // Try project root next (usually from within a Sandbox Lambda)
-  if (!result.filepath) result = readArc({ cwd: join(process.cwd(), '..', '..', '..') })
-  // If we still didn't find it, blow up
-  if (!result.filepath) throw ReferenceError('Architect project manifest not found')
+    try {
+      // attempt to read the arcfile
+      let { arc } = read({ type, cwd })
 
-  let { arc } = result
-  if (arc.tables && arc.tables.some(t => Object.keys(t)[0] === 'data'))
-    return `${arc.app[0]}-staging-data`
-  throw ReferenceError('@tables not defined! begin/data expects a data table')
-  */
+      // look for a table named 'data'
+      if (arc.tables && arc.tables.some(t => Object.keys(t)[0] === 'data')) {
```

Comment:
If we have Inventory in Begin Data, why using the manifest reader? Imo best practice here would be to run Inventory.

Reply:
Good question!
I think there was a reason @brianleroux needed to just use `read`